File: week-1---cplusplus-intro/lab_reverse.py
def reverse( sequence ):
	reversed = []
	for i in range(len(sequence)-1,-1,-1):
		reversed.append( sequence[i] )
	return reversed

if __name__ == "__main__":
	sequence = [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 ]

	reversed = reverse( sequence )

	print( "original: ", end="" )
	for i in range(len(sequence)):
		print( sequence[i], end=", " )
	print()

	print( "reversed: ", end="" )
	for i in range(len(reversed)):
		print( reversed[i], end=", " )
	print()

File: week-1---cplusplus-intro/lab_reverse.cpp
#include "lab_reverse.h"

#include <iostream>
#include <vector>

int main()
{
	std::vector<int> sequence { 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };

	std::vector<int> reversed = reverse( sequence );

	std::cout << "original: ";
	for( int i=0; i<sequence.size(); ++i )
		std::cout << sequence[i] << ", ";
	std::cout << std::endl;

	std::cout << "reversed: ";
	for( int i=0; i<reversed.size(); ++i )
		std::cout << reversed[i] << ", ";
	std::cout << std::endl;

	return 0;
}
